app/models/amusement_park.rb

```diff
@@ -1,3 +1,7 @@
 class AmusementPark < ApplicationRecord
   has_many :rides
+
+  def mechanics_working_at_park
+    Mechanic.joins(rides: :amusement_park).where("amusement_parks.id = #{self.id}").distinct
```

Comment:
Nice joining! An even easier option for this is to leverage the association already set up for rides and add:
`has_many :mechanics, through: :rides`

With this additional association, you can replace this method with: `mechanics.distinct` !!!